app/src/main/java/org/mozilla/tv/firefox/pocket/PocketRepo.kt

```diff
@@ -0,0 +1,106 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.tv.firefox.pocket
+
+import android.arch.lifecycle.LiveData
+import android.arch.lifecycle.MutableLiveData
+import android.os.SystemClock
+import android.support.annotation.UiThread
+import kotlinx.coroutines.experimental.CancellationException
+import kotlinx.coroutines.experimental.DefaultDispatcher
+import kotlinx.coroutines.experimental.Job
+import kotlinx.coroutines.experimental.delay
+import kotlinx.coroutines.experimental.launch
+import kotlinx.coroutines.experimental.withContext
+import org.mozilla.tv.firefox.BuildConfig
+import java.util.concurrent.TimeUnit
+
+private val CACHE_UPDATE_FREQUENCY_MILLIS = TimeUnit.MINUTES.toMillis(45)
+
+sealed class PocketRepoState {
```

Comment:
nit: `PocketVideoRepo.FeedState`? I think it's unclear what "state" will represent.